drivers/hwinfo/hwinfo_ambiq.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2024 Ambiq
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <am_mcu_apollo.h>
+#include <zephyr/drivers/hwinfo.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+
+struct ambiq_hwinfo {
+	uint32_t ui32ChipID0;
+	uint32_t ui32ChipID1;
+	uint32_t ui32TrimVer;
+};
+
+struct ambiq_hwinfo sAmbiqHwInfo = {0};
```

Comment:
Please put this struct inside the function and use the Zephyr naming conventions.

Reply:
Will do